RecoMTD/TrackExtender/plugins/TrackExtenderWithMTD.cc

```diff
@@ -494,83 +940,41 @@ reco::Track TrackExtenderWithMTDT<TrackCollection>::buildTrack(const reco::Track
-                     ForwardSubdetector(it->recHit()->geographicalId().subdetId()) == FastTime;
-        if (ismtd) {
-          const auto& propresult2 = thePropagator->propagateWithPath(
-              tscbl.trackStateAtPCA(), trajWithMtd.firstMeasurement().updatedState().surface());
-          pathlength2 = propresult2.second;
-          if (pathlength2 == 0.) {
-            validpropagation = false;
-          }
-          pathlength = pathlength1 + pathlength2;
-          const MTDTrackingRecHit* mtdhit = static_cast<const MTDTrackingRecHit*>(it->recHit()->hit());
-          thit = mtdhit->time();
-          thiterror = mtdhit->timeError();
-          validmtd = true;
-          break;
-        }
-      }
-    } else {
-      for (auto it = trajWithMtd.measurements().rbegin(); it != trajWithMtd.measurements().rend(); ++it) {
-        bool ismtd = it->recHit()->geographicalId().det() == DetId::Forward &&
-                     ForwardSubdetector(it->recHit()->geographicalId().subdetId()) == FastTime;
-        if (ismtd) {
-          const auto& propresult2 = thePropagator->propagateWithPath(
-              tscbl.trackStateAtPCA(), trajWithMtd.lastMeasurement().updatedState().surface());
-          pathlength2 = propresult2.second;
-          if (pathlength2 == 0.) {
-            validpropagation = false;
-          }
-          pathlength = pathlength1 + pathlength2;
-          const MTDTrackingRecHit* mtdhit = static_cast<const MTDTrackingRecHit*>(it->recHit()->hit());
-          thit = mtdhit->time();
-          thiterror = mtdhit->timeError();
-          validmtd = true;
-          break;
-        }
+
+    //need to better handle the cases with >1 hit in MTD
+    for (auto it = trajWithMtd.measurements().begin(); it != trajWithMtd.measurements().end(); ++it) {
+      bool ismtd = it->recHit()->geographicalId().det() == DetId::Forward &&
+                   ForwardSubdetector(it->recHit()->geographicalId().subdetId()) == FastTime;
+      if (ismtd) {
+        const MTDTrackingRecHit* mtdhit = static_cast<const MTDTrackingRecHit*>(it->recHit()->hit());
+        thit = mtdhit->time();
+        thiterror = mtdhit->timeError();
+        validmtd = true;
+        break;
       }
     }
 
     if (validmtd && validpropagation) {
-      double magp2 = p.mag2();
-
-      double gammasq_pi = 1. + magp2 * m_pi_inv2;
-      double beta_pi = std::sqrt(1. - 1. / gammasq_pi);
-      double dt_pi = pathlength / beta_pi * c_inv;
-
-      double gammasq_p = 1. + magp2 * m_p_inv2;
-      double beta_p = std::sqrt(1. - 1. / gammasq_p);
-      double dt_p = pathlength / beta_p * c_inv;
-
-      double dterror = dt_p - dt_pi;
-      double betaerror = beta_p - beta_pi;
-
+      TrackTofPidInfo tofInfo(
+          p.mag2(),
+          pathlength,
+          thit,
+          thiterror,
+          0.,
+          0.,
+          true);                   //here add the PID uncertainty for later use in the 1st step of 4D vtx reconstruction
```

Comment:
this looks like a case of clang-format breaking up the code lines for no reason.
Please move the comment above ```TrackTofPidInfo tofInfo```